src/plugins/fs/reiserfs.c

```diff
@@ -0,0 +1,386 @@
+
+    success = get_uuid_label (device, &(ret->uuid), &(ret->label), error);
+    if (!success) {
+        /* error is already populated */
+        bd_fs_reiserfs_info_free (ret);
+        return NULL;
+    }
+
+    lines = g_strsplit (output, "\n", 0);
+    g_free (output);
+    line_p = lines;
+    /* find the beginning of the (data) section we are interested in */
+    while (line_p && *line_p && !g_str_has_prefix (*line_p, "Count of blocks on the device:"))
+        line_p++;
+    if (!line_p || !(*line_p)) {
+        g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_PARSE, "Failed to parse ReiserFS file system information");
+        g_strfreev (lines);
+        bd_fs_reiserfs_info_free (ret);
+        return FALSE;
+    }
+
+    /* extract data from something like this: "Count of blocks on the device: 127744" */
+    val_start = strchr (*line_p, ':');
+    val_start++;
+    ret->block_count = g_ascii_strtoull (val_start, NULL, 0);
+
+    while (line_p && *line_p && !g_str_has_prefix (*line_p, "Blocksize:"))
+        line_p++;
+    if (!line_p || !(*line_p)) {
+        g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_PARSE, "Failed to parse ReiserFS file system information");
+        g_strfreev (lines);
+        bd_fs_reiserfs_info_free (ret);
+        return FALSE;
+    }
+
+    /* extract data from something like this: "Blocksize: 4096" */
+    val_start = strchr (*line_p, ':');
+    val_start++;
+    ret->block_size = g_ascii_strtoull (val_start, NULL, 0);
+
+    while (line_p && *line_p && !g_str_has_prefix (*line_p, "Free blocks"))
+        line_p++;
+    if (!line_p || !(*line_p)) {
+        g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_PARSE, "Failed to parse ReiserFS file system information");
+        g_strfreev (lines);
+        bd_fs_reiserfs_info_free (ret);
+        return FALSE;
+    }
+
+    /* extract data from something like this: "Free blocks (count of blocks - used [journal, bitmaps, data, reserved] blocks): 119529" */
+    val_start = strchr (*line_p, ':');
+    val_start++;
+    ret->free_blocks = g_ascii_strtoull (val_start, NULL, 0);
+
+    return ret;
+}
+
+/**
+ * bd_fs_reiserfs_resize:
+ * @device: the device the file system of which to resize
+ * @new_size: new requested size for the file system (if 0, the file system is
+ *            adapted to the underlying block device)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether the file system on @device was successfully resized or not
+ *
+ * Tech category: %BD_FS_TECH_REISERFS-%BD_FS_TECH_MODE_RESIZE
+ */
+gboolean bd_fs_reiserfs_resize (const gchar *device, guint64 new_size, GError **error) {
+    const gchar *args[6] = {"fsadm", "-y", "resize", device, NULL, NULL};
+    gboolean ret = FALSE;
+    BDFSReiserFSInfo *info = NULL;
+
+    if (!check_deps (&avail_deps, DEPS_RESIZEREISERFS_MASK, deps, DEPS_LAST, &deps_check_lock, error) ||
+        !check_deps (&avail_deps, DEPS_DEBUGREISERFS_MASK, deps, DEPS_LAST, &deps_check_lock, error))
```

Comment:
`DEPS_DEBUGREISERFS_MASK` is checked by `bd_fs_reiserfs_get_info()`, right?

Reply:
Good point, fixed.